lib/palindrome_permutation.rb

```diff
@@ -1,4 +1,21 @@
 
 def palindrome_permutation?(string)
-  raise NotImplementedError, "palindrome_permutation? not implemented"
+  letter_hash = {}
+
+  string.split("").each do |letter|
+    if letter_hash[letter] == nil
+      letter_hash[letter] = 1
+    else
+      letter_hash[letter] += 1
+    end
+  end
+
+  midpoint = 0
+  letter_hash.each_value do |times|
+    if times % 2 != 0
+      midpoint += 0
+    end
+  end
```

Comment:
You would do better here to see how many letters appear an odd number of times.

Reply:
Sorry about that, Chris! I had a typo in my code. so it should be 
`midpoint += 1` instead of `midpoint += 0` 
I run rake file after that seems to fix the failures, would that be consider working for this method? Just wondering. Thanks!

`
def palindrome_permutation?(string)
  letter_hash = {}

  string.split("").each do |letter|
    if letter_hash[letter] == nil
      letter_hash[letter] = 1
    else
      letter_hash[letter] += 1
    end
  end

  midpoint = 0
  letter_hash.each_value do |times|
    if times % 2 != 0
      midpoint += 1
    end
  end
  return midpoint <= 1
end
`